syntax="proto3";

package executer;

// Interface exported by the executer node and exposed to end user (very simple so far).
service Executer {
    rpc ShortestPathQuery(QueryData) returns (QueryFinished);

    rpc HealthCheck(HealthCheckRequest) returns (HealthCheckResponse);
}

message QueryData {
    uint64 node_id_from = 1;
    uint64 node_id_to = 2;
}

// TODO: Add backtracing and send the full path back
message QueryFinished {
    // Not present if the path was not found
    optional uint64 shortest_path_len = 1;
}

message HealthCheckRequest {}

message HealthCheckResponse {
    enum ServingStatus {
        UNKNOWN = 0;
        SERVING = 1;
        NOT_SERVING = 2;
    }
    ServingStatus status = 1;
}